import "common.proto";
import "auth.proto";
import "collection.proto";
import "media.proto";

package Remote;

message ClientMessage
{
	enum Type
	{
		AuthRequest		= 1;
		AudioCollectionRequest	= 2;
		MediaRequest		= 3;
	}

	required Type	type	= 1;

	optional AuthRequest		auth_request			= 2;
	optional AudioCollectionRequest	audio_collection_request	= 3;
	optional MediaRequest		media_request			= 4;
}


message ServerMessage
{
	enum Type
	{
		AuthResponse		= 1;
		AduioCollectionResponse	= 2;
		MediaResponse		= 3;
	}

	required Type	type	= 1;

	optional AuthResponse			auth_response			= 2;
	optional AudioCollectionResponse	audio_collection_response	= 3;
	optional MediaResponse			media_response			= 4;
}


